securedrop_client/gui/widgets.py

```diff
@@ -2007,351 +2009,538 @@ def _on_left_click(self):
+        font-family: 'Montserrat';
+        font-size: 16px;
+        color: #302aa3;
+        padding-bottom: 20px;
+    }
+    #button_box QPushButton {
+        margin: 0px 0px 0px 12px;
+        height: 40px;
+        padding-left: 20px;
+        padding-right: 20px;
+        border: 2px solid #2a319d;
+        font-family: 'Montserrat';
+        font-weight: 500;
+        font-size: 15px;
+        color: #2a319d;
+    }
+    #button_box QPushButton::disabled {
+        border: 2px solid rgba(42, 49, 157, 0.4);
+        color: rgba(42, 49, 157, 0.4);
+    }
+    #button_box QPushButton#primary_button {
+        background-color: #2a319d;
+        color: #fff;
+    }
+    #button_box QPushButton#primary_button::disabled {
+        border: 2px solid #C2C4E3;
+        background-color: #C2C4E3;
+        color: #E1E2F1;
     }
     '''
 
-    def __init__(self, controller, file_uuid):
-        super().__init__()
+    CONTENT_MARGIN = 40
+    HEADER_MARGIN = 0
+    BODY_MARGIN = 0
 
-        self.controller = controller
-        self.file_uuid = file_uuid
+    dialog_closing = pyqtSignal()
+
+    def __init__(self):
+        parent = QApplication.activeWindow()
+        super().__init__(parent)
 
-        self.setObjectName('print_dialog')
+        self.setObjectName('frameless_dialog')
         self.setStyleSheet(self.CSS)
-        self.setWindowFlags(Qt.Popup)
-        self.setWindowModality(Qt.WindowModal)
+        self.setWindowFlags(Qt.Widget)
 
+        # Set drop shadow effect
+        effect = QGraphicsDropShadowEffect(self)
+        effect.setOffset(0, 1)
+        effect.setBlurRadius(8)
+        effect.setColor(QColor('#aa000000'))
+        self.setGraphicsEffect(effect)
+        self.update()
+
+        # Custom titlebar for close button
+        titlebar = QWidget()
+        titlebar_layout = QVBoxLayout()
+        titlebar.setLayout(titlebar_layout)
+        close_button = SvgPushButton('delete_close.svg', svg_size=QSize(10, 10))
+        close_button.setObjectName('close_button')
+        close_button.setText('CLOSE')
+        close_button.clicked.connect(self.close)
+        titlebar_layout.addWidget(close_button, alignment=Qt.AlignRight)
+
+        # Content including: header, body, help menu, and buttons
+        content = QWidget()
+        content_layout = QVBoxLayout()
+        content.setLayout(content_layout)
+        content_layout.setContentsMargins(
+            self.CONTENT_MARGIN, 0, self.CONTENT_MARGIN, self.CONTENT_MARGIN)
+        header_container = QWidget()
+        header_container_layout = QHBoxLayout()
+        header_container_layout.setContentsMargins(
+            self.HEADER_MARGIN, self.HEADER_MARGIN, self.HEADER_MARGIN, self.HEADER_MARGIN)
+        header_container.setLayout(header_container_layout)
+        self.header_icon = SvgLabel('blank.svg', svg_size=QSize(64, 64))
+        self.header_icon.setObjectName('header_icon')
+        self.header_icon.setFixedWidth(80)
+        self.header = QLabel()
```

Comment:
in the child classes here we're using this widget to display the filename which is user-controlled input, so this should be SecureQLabel

Reply:
`SecureQLabel` gets rid of the formatting needed to make the header look like the zeplin spec unfortunately. The header would need to be updated to be two SecureQLabels because there are two different styles in the same header: one for the "Task name" the other for the filename.